src/graphnet/models/task/task.py

```diff
@@ -38,6 +55,7 @@ def __init__(self, hidden_size: int, target_label: str, loss_function: LossFunct
     def forward(self, x: Union[Tensor, Data]) -> Union[Tensor, Data]:
         self._regularisation_loss = 0  # Reset
         x = self._affine(x)
+        x = self._transform_forward(x)
         return self._forward(x)
 
```

Comment:
```suggestion
    @final
    def _transform_prediction(self, prediction: Union[Tensor, Data]) -> Union[Tensor, Data]:
        if self._inference:
            return self._transform_prediction_inference(prediction)
        else:
            return self._transform_prediction_training(prediction)
```

Reply:
Oops, the `self._transform_forward` should be renamed to `self._transform_prediction`, or the other way around, if the suggested change is accepted. Otherwise the names would disagree.